enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+  installer. By making the definition of the content and configuration
+  of the components in a single node developer cluster more orthogonal
+  to the tooling that deploys the cluster, this enhancement could
+  enable distinct future enhancements to improve the build and/or
+  install process used by CRC.
+* A new OpenShift installer. This enhancement proposal does not add a
+  new OpenShift installer or involve changes, beyond the ability to
+  specify the cluster profile, to the standard OpenShift installer
+  currently used to build CRC installation images. Evolution of the
+  existing installer, SNC tooling and other CRC components is
+  expected.
+* Functional changes for OpenShift. This enhancement proposal does not
+  result in any functional changes to clusters that do not utilize the
+  single-node-developer cluster profile.
+
+## Proposal
+
+The single-node-developer cluster profile will be implemented in two
+phases. The first phase introduces the new cluster profile, and
+enables some simplification of the SNC image build process by
+utilizing the profile to control which components are included in the
+cluster. The second phase focuses on optimizing the resource
+utilization of the components included in the cluster and may further
+simplify the SNC image build process.
+
+### Phase One
+
+In the initial phase, the following annotation will be added to the
+manifest files for all components that are to be included in single
+node developer clusters:
+
+```
+include.release.openshift.io/single-node-developer=true
+```
+
+This annotation will need to be added to the manifests for all the
+OpenShift components that are not currently removed by the snc.sh
+script during the CRC build process. The components that are currently
+removed, and thus should not need the annotation added, include:
+
+* most operators running in the openshift-monitoring namespace
+* most operators running in the openshift-insights namespace
+* machine-api-operator
+* Machine-config-operator (may need to run once, requiring the annotation)
+* openshift-cloud-credential-operator
+* openshift-cluster-storage-operator
+
+Additional components that are not currently removed by the SNC
+tooling might be also be determined to be unnecessary, and thus not
+require the annotation. Possibilities include:
+
+* openshift-cluster-node-tuning-operator
+* openshift-multus
+
+Once PRs adding this annotation have been merged to all the necessary
+component repositories, the snc.sh script will be updated to tell the
+OpenShift installer to use the single-node-developer cluster profile,
+and to remove the code that deletes the components that are not
+annotated. At the conclusion of this initial phase, the somewhat
+simplified CRC build tooling should produce a cluster equivalent to
+that currently produced, and all existing CRC tests should pass. Phase
+one is expected to be completed during development of the 4.7
+OpenShift release.
```

Comment:
Does this mean other teams implicitly got another epic to support this effort in 4.7 ?

Reply:
Phase one changes would most likely be similar to https://github.com/openshift/cluster-authentication-operator/pull/352 , one line addition to manifests which need to be included, not sure this would use so much time?

Reply:
Phase one is fine, if these annotation are enough to get it working. I am more worried about https://github.com/openshift/enhancements/pull/482#discussion_r503152485.

Reply:
For the 4.7 timeframe, I don't expect there will be more than 'phase one' happening, ie adding the relevant annotation to manifests. At that stage, not having a way to use different manifest for operands will hopefully be good enough. Once this 'phase one' single node profile is in place, we can start narrowing down which components need profile-specific manifests, and scope/plan the work with the relevant teams.

Reply:
I agree completing phase 1 is the main goal for this enhancement in 4.7, but hope we'll be able to also include specialized manifests for at least one or two components, such as etcd-operator, in that release.

Reply:
@sttts please see my earlier comments around shared responsibility versus ownership.  I see alternative manifests as a shared responsibility across the teams, but _ownership_ in 4.7 for any alternative manifests is with CRC.  This is similar to how we handled ROKS.  It is not the intention that every team got a new epic.